src/main/java/com/sumologic/log4j/core/SumoJsonLayout.java

```diff
@@ -0,0 +1,232 @@
+import org.apache.logging.log4j.core.impl.Log4jLogEvent;
+import org.apache.logging.log4j.core.impl.ThrowableProxy;
+import org.apache.logging.log4j.core.jackson.JsonConstants;
+import org.apache.logging.log4j.core.jackson.Log4jJsonObjectMapper;
+import org.apache.logging.log4j.core.layout.AbstractStringLayout;
+import org.apache.logging.log4j.core.util.StringBuilderWriter;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.util.Strings;
+
+/**
+ * Borrowed heavily from org.apache.logging.log4j.core.layout.JsonLayout
+ */
+@Plugin(name = "SumoJsonLayout", category = Node.CATEGORY, elementType = Layout.ELEMENT_TYPE, printObject = true)
+public class SumoJsonLayout extends AbstractStringLayout
+{
+  private static final String CONTENT_TYPE = "application/json; charset=" + StandardCharsets.UTF_8.displayName();
+
+  private final ObjectWriter objectWriter;
+
+  @PluginFactory
+  public static SumoJsonLayout createLayout(
+      // @formatter:off
+      @PluginAttribute(value = "locationInfo", defaultBoolean = false) final boolean locationInfo
+      // @formatter:on
+  )
+  {
+    final SimpleFilterProvider filters = new SimpleFilterProvider();
+    final Set<String> except = new HashSet<>(2);
+    if (!locationInfo) {
+      except.add(JsonConstants.ELT_SOURCE);
+    }
+    except.add("loggerFqcn");
+    except.add("endOfBatch");
+    except.add(JsonConstants.ELT_NANO_TIME);
+    filters.addFilter(Log4jLogEvent.class.getName(), SimpleBeanPropertyFilter.serializeAllExcept(except));
+    final ObjectWriter writer = new Log4jJsonObjectMapper().writer(new MinimalPrettyPrinter());
+    return new SumoJsonLayout(writer.with(filters));
+  }
+
+  SumoJsonLayout(ObjectWriter objectWriter)
+  {
+    super(StandardCharsets.UTF_8, null, null);
+    this.objectWriter = objectWriter;
+  }
+
+  /**
+   * Formats a {@link org.apache.logging.log4j.core.LogEvent}.
+   *
+   * @param event The LogEvent.
+   *
+   * @return The JSON representation of the LogEvent.
+   */
+  @Override
+  public String toSerializable(final LogEvent event)
+  {
+    final StringBuilderWriter writer = new StringBuilderWriter();
+    try {
+      objectWriter.writeValue(writer, wrap(event));
+      writer.write('\n');
+      return writer.toString();
+    }
+    catch (final IOException e) {
+      LOGGER.error(e);
+      return Strings.EMPTY;
+    }
+  }
+
+  // Overridden in tests
+  LogEvent wrap(LogEvent event)
+  {
+    return new SumoLogEvent(event);
+  }
+
+  @Override
+  public String getContentType()
+  {
+    return CONTENT_TYPE;
+  }
+}
+
+// Jackson and SumoLogic friendly class
+class SumoLogEvent implements LogEvent
+{
+  private final LogEvent delegate;
+
+  SumoLogEvent(LogEvent delegate)
+  {
+    this.delegate = delegate;
+  }
+
+  @Override
+  public Map<String, String> getContextMap()
+  {
+    final Map<String, String> map = delegate.getContextMap();
+    return map.isEmpty() ? null : map;
```

Comment:
[`LogEvent.getContextMap()`](https://logging.apache.org/log4j/2.0/log4j-core/apidocs/org/apache/logging/log4j/core/LogEvent.html#getContextMap()) Javadoc says that this method should never return `null`.